lib/kafka/produce_operation.rb

```diff
@@ -137,6 +137,9 @@ def handle_response(response)
               delay: ack_time - message.create_time,
             })
           end
+        rescue Kafka::MessageSizeTooLarge
+          @logger.error "Message is too large when writing to #{topic}/#{partition}"
+          @buffer.clear_messages(topic: topic, partition: partition)
```

Comment:
At any rate, the policy should be set at the producer level, not the operation level. I think it's preferable to raise the error further up that stack and handle it specifically in AsyncProducer, which doesn't guarantee delivery anyway. The sync producer should never silently lose data.